### PR TITLE
Refactored code and added perf traces and telemetry

### PR DESCRIPTION
**Pull request by ShreyasRmsft**

### CODE REVIEW

File: Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Common/SimpleTimer.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    /// }
+    /// </summary>
+    public class SimpleTimer : IDisposable
+    {
+        /// <summary>
+        /// Creates a timer with threshold. A perf message is logged only if
+        /// the time elapsed is more than the threshold.
+        /// </summary>
+        public SimpleTimer(string timerName, string telemetryArea, string telemetryEventName, ITraceLogger logger, 
+            ITelemetryDataCollector telemetryDataCollector, TimeSpan threshold, bool publishTelemetry = true)
```
Comment (acesiddhu) [inline Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Common/SimpleTimer.cs]: > bool publishTelemetry = true [](start = 79, length = 29)

remove this param and always do it

File: Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Common/SimpleTimer.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                return;
+            }
+
+            if (_timer.Elapsed > _threshold)
+            {
+                _logger.Warning($"PERF : {_name} : took {_timer.Elapsed.TotalMilliseconds} ms.");
+                if (_publishTelemetry)
+                {
+                    // Is the precision enough?
+                    _telemetry.AddToCumulativeTelemetry(_telemetryArea, _telemetryEventName, _timer.Elapsed, true);
```
Comment (acesiddhu) [inline Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Common/SimpleTimer.cs]: >Elapsed [](start = 100, length = 7)

Is timestamp serailizable

Comment (acesiddhu) [reply Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Common/SimpleTimer.cs]: log threshold also

---
In reply to: [251381664](https://github.com/nigurr/TestResultParser/pull/46#discussion_r251381664) [](ancestors = 251381664)

File: Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Common/SimpleTimer.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+            if (disposing)
+            {
+                StopAndLog();
+                GC.SuppressFinalize(this);
```
Comment (acesiddhu) [inline Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Common/SimpleTimer.cs]: > GC.SuppressFinalize(this) [](start = 15, length = 26)

remove this

Comment (ShreyasRmsft) [reply Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Common/SimpleTimer.cs]: done

File: Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Jasmine/JasmineRegexes.cs [hunk 4]
```diff
@@ -3,56 +3,60 @@
...
     using System.Text.RegularExpressions;
 
     public class JasmineRegexes
     {
         /// <summary>
-        /// Matches lines with the following regex:
-        /// ^Started$
+        /// Default timeout for all the regexes
         /// </summary>
-        public static Regex TestRunStart { get; } = new Regex($"^Started$");
```
Comment (ShreyasRmsft) [inline Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Jasmine/JasmineRegexes.cs]: one extra property was added, wierdly git is showing each line as changed

**Pull request merged**